backend/controllers/appointmentConfirmDeletion.js

```diff
@@ -0,0 +1,86 @@
+const packAppointmentDetails = (req, details) => {
+  req.flash('appointmentDetails', details)
+}
+
+const unpackAppointmentDetails = req => {
+  const appointmentDetails = req.flash('appointmentDetails')
+  if (!appointmentDetails || !appointmentDetails.length) throw new Error('Appointment details are missing')
+
+  return appointmentDetails.reduce(
+    (acc, current) => ({
+      ...acc,
+      ...current,
+    }),
+    {}
+  )
+}
+
+const clearAppointmentDetails = req => {
+  req.flash('appointmentDetails')
+}
+
+module.exports = ({ whereaboutsApi }) => {
+  const renderTemplate = (req, res, errors) => {
+    const appointmentDetails = unpackAppointmentDetails(req)
+    const {
+      id,
+      isRecurring,
+      additionalDetails,
+      basicDetails,
+      prepostData,
+      recurringDetails,
+      timeDetails,
+    } = appointmentDetails
+
+    // Save for possible re-showing on error
+    packAppointmentDetails(req, appointmentDetails)
+
+    return res.render('appointmentConfirmDeletion', {
+      errors,
+      appointmentEventId: id,
+      isRecurring,
+      additionalDetails,
+      basicDetails,
+      prepostData,
+      recurringDetails,
+      timeDetails,
+    })
+  }
+
+  const index = async (req, res) => {
+    try {
+      renderTemplate(req, res, [])
+    } catch (error) {
+      res.locals.redirectUrl = `/view-all-appointments`
+      throw error
+    }
+  }
+
+  const post = async (req, res) => {
+    const { confirmation, isRecurring, appointmentEventId } = req.body
+
+    if (!confirmation) {
+      const errors = []
+      errors.push({ text: 'Select yes if you want to delete this appointment', href: '#confirmation' })
+      return renderTemplate(req, res, errors)
```

Comment:
We should be doing a redirect back to the GET route here. The pattern is called PRG (Post, redirect, get)

Reply:
Hmm.. I copied what was done elsewhere (e.g. bulkAppointmentsConfirm).

I also don't know why it is necessary?!

Reply:
It's standard practice to stop the form from being resubmitted on page refresh.

Reply:
Standard practice that we don't do everywhere?!

Reply:
I may of missed something. Can we have a call as it feels odd doing it all on here.

Reply:
We will change this, as showing info in a post request leads to problems if the user refreshes the page